amd_openvx_extensions/amd_nn/src/slice_layer.cpp

```diff
@@ -1,345 +1,386 @@
+    else {
+        for(int i=0; i<param_dims; i++)
+            steps.push_back(1);
+    }
+
+    // calculate and create indices tensor
+    std::vector<std::vector<int>> indices(param_dims);
+    for(int i=0; i<param_dims; i++) {
+        int index = starts[i];
+        while (index < ends[i]) {
+            indices[i].push_back(index);
+            index += steps[i];
+        }
+
+        vx_size dim[1] = {indices[i].size()};
+        
+        vx_tensor tmp_tensor = vxCreateTensor(context, 1, dim, type, 0);
+        status = vxMapTensorPatch(tmp_tensor, 1, nullptr, nullptr, &map_id, stride, (void **)&ptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST, 0);
+        if(status) {
+            std::cerr << "ERROR: vxMapTensorPatch() failed for indices tensor(" << status << ")" << std::endl;
+            return -1;
+        }
+        
+        for(auto itr = indices[i].begin(); itr != indices[i].end(); itr++) {
+            *ptr++ = *itr;
+        }
+        status = vxUnmapTensorPatch(tmp_tensor, map_id);
+        indices_tensor[i] = tmp_tensor;
+    }
+
+    // calculate output dims and create output tensor
+    vx_size tmp_num_dims = input_num_dims;
+    vx_size *tmp_dims = input_dims;
+
+    for(int i=0; i<(param_dims-1); i++) {
+        //reverse input dims w,h,c,n- > n,c,h,w
+        int start = 0, end = tmp_num_dims-1;
+        while (start < end) {
+            int temp = tmp_dims[start];
+            tmp_dims[start] = tmp_dims[end];
+            tmp_dims[end] = temp;
+            start++;
+            end--;
+        }
+
+        int out_dim_rank = tmp_num_dims;    
+        vx_size out_dims[out_dim_rank];
+        
+        for (int j=0; j<out_dim_rank; j++) {
+            if (j == axes[i]) {
+                out_dims[j] = indices[i].size();
+            }
+            else {
+                out_dims[j] = tmp_dims[j];
+            }
+        }
+
+        //reverse output dims n,c,h,w -> w,h,c,n
+        start = 0, end = out_dim_rank-1;
+        while (start < end) {
+            int temp = out_dims[start];
+            out_dims[start] = out_dims[end];
+            out_dims[end] = temp;
+            start++;
+            end--;
+        }
+        
+        //vx_tensor tmp_tensor = vxCreateTensor(context, out_dim_rank, out_dims, input_type, 0);
+        vx_tensor tmp_tensor = vxCreateVirtualTensor(graph, out_dim_rank, out_dims, input_type, 0);
+        input_tensor[i+1] = tmp_tensor;
+        output_tensor[i] = tmp_tensor;
+        
+        tmp_num_dims = out_dim_rank;
+        tmp_dims = out_dims;
+    }
+    
+    // create temp output tensor which will hold the final output result
+    vx_tensor tmpo_tensor = vxCreateTensor(context, output_num_dims, output_dims, input_type, 0);
+    output_tensor[param_dims-1] = tmpo_tensor;
+
+    for (int i=0; i<param_dims; i++) {
+        vxGatherLayer(graph, (vx_tensor)input_tensor[i], (vx_tensor)indices_tensor[i], (vx_tensor)output_tensor[i], axes[i]);
+    }
+    
+    status = vxVerifyGraph(graph);
```

Comment:
Codacy found an issue: [Variable 'status' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5932170)